### PR TITLE
Update software for edition 2018.

### PR DESCRIPTION
**Pull request by brandonedens**

### CODE REVIEW

**Comment by nastevens** [comment]
This will increase the minimum supported Rust version to 1.31. Could you please update the README and Travis CI settings?

File: README.md [hunk 1]
```diff
@@ -76,8 +76,8 @@ The following features are implemented and planned for the library:
 
 ## Minimum Supported Rust Version (MSRV)
 
-This crate is guaranteed to compile on stable Rust 1.26 and up.  It *might*
-compile with older versions but that may change in any new patch release.
+This crate is guaranteed to compile on stable Rust 1.31 and up (as it requires
+marked as edition 2018).
```
Comment (nastevens) [inline README.md]: No need to change the wording here - this is a standard bit of boilerplate for rust-embedded projects. Just update the number.

Comment (brandonedens) [reply README.md]: Fixed.

**Review by posborne — APPROVED** [review]
Seems fine.  The MSRV change is breaking and requires a version bump (0.5.0) as per MSRV guidelines: https://github.com/rust-embedded/wg/blob/master/ops/msrv.md#minimum-supported-rust-version-msrv

I'm fine merging as-is; there's probably a couple other changes that might go in prior to a 0.5.0 release so holding this change up on that isn't critical and just needs to be addressed prior to release or as part of the release PR/Changelog update.

**Pull request merged**